src/PhpSpreadsheet/Writer/Xlsx/StringTable.php

```diff
@@ -42,35 +29,33 @@ public function createStringTable(Worksheet $pSheet, $pExistingTable = null)
                 !is_object($cellValue) &&
                 ($cellValue !== null) &&
                 $cellValue !== '' &&
-                ($cell->getDataType() == DataType::TYPE_STRING || $cell->getDataType() == DataType::TYPE_STRING2 || $cell->getDataType() == DataType::TYPE_NULL) &&
-                !isset($aFlippedStringTable[$cellValue])
+                !$aFlippedStringTable->has(md5($cellValue)) &&
```

Comment:
Hello, 

I've being trying to use this branch and noticed that my unit tests writing and reading XLSX were failing with an index out of bound error. 

Drilling through, I eventually found out that the "cache" (in the unit test  a shared array in memory, but in production a shared redis), would acttually potentially clash when writing multiple XLSX file that re-use string between the spreadsheet eg:

- because the cache exists, it would not write the text in the cache and increment the stringTableRecordsCount value. (the sharedStrings.xml would not contain the entry either)
- when writing the sheet.xml it would then find the entry in the cache again, and point to a non existent offset.

in my branch, I've been generating a 'uniqid()' string to avoid any cache clash and pass it around the same way the CacheInterface is passed around. (and the cache will eventually purge itself anyway per cache underlying configuration).